src/features/init.rs

```diff
@@ -0,0 +1,153 @@
+// Local project operations and local file configuration handlers
+
+use error_stack::{IntoReport, Result, ResultExt};
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::{error::Error, fmt::Display};
+
+// Error ProjectConfig
+#[derive(Debug)]
+pub enum ProjectConfigError {
+    InvalidPath,
+    InvalidConfig,
+}
+
+impl Display for ProjectConfigError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ProjectConfigError::InvalidPath => write!(f, "Invalid path"),
+            ProjectConfigError::InvalidConfig => write!(f, "Invalid config"),
+        }
+    }
+}
+
+impl Error for ProjectConfigError {}
+
+// Struct to handle the project configuration
+#[derive(Serialize, Deserialize)]
+pub struct ProjectConfig {
+    pub identifier: String,
+    pub url: String,
+    pub branch_kinds: Vec<BranchKind>,
+}
+
+// Struct to handle branch kinds
+// Default options are:
+// Feature,
+// Bugfix,
+// Release,
+// Hotfix,
+// Support,
+// Other,
+#[derive(Serialize, Deserialize)]
+pub struct BranchKind {
+    pub kind: String,
+    pub formatter: String,
+}
+
+// Function with the configuration file path
+fn get_local_config_file() -> Result<File, ProjectConfigError> {
+    let mut path = std::env::current_dir()
+        .into_report()
+        .change_context(ProjectConfigError::InvalidPath)?;
+    path.push(".gbc"); // JSON formatted local file
```

Comment:
It's weird to me that `path` is mutable just to have the file added (but it can be the best way of doing this?).

Reply:
It's like that because I was thinking to add some flag to the command later so you can change the name of the file but also looks like is faster that string interpolation (or some weird rust thing that I have 0 knowledge but looks like is common)